<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LocationWidget</class>
 <widget class="QWidget" name="LocationWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>690</width>
    <height>995</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QPushButton" name="m_btn_parent">
       <property name="text">
        <string>Parent</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Child locations count:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="m_child_location_count">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>[VALUE]</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QListWidget" name="m_child_locations"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Local position:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="m_local_x"/>
         </item>
         <item>
          <widget class="QLineEdit" name="m_local_y"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Global position:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="m_global_x">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="m_global_y">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Rotation:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="m_rotation_angle"/>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0" columnminimumwidth="0,0,0">
         <property name="spacing">
          <number>0</number>
         </property>
         <item row="3" column="1">
          <widget class="QPushButton" name="m_btn_move_bottom">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
             <horstretch>1</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="baseSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="text">
            <string>Bottom</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="m_btn_move_left">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
             <horstretch>1</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Left</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="m_btn_move_right">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
             <horstretch>1</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Right</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="m_btn_move_top">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
             <horstretch>1</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Top</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QPushButton" name="m_btn_rot_left">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
             <horstretch>1</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Rotate left</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="m_btn_rot_right">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
             <horstretch>1</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Rotate Right</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>m_btn_move_bottom</sender>
   <signal>clicked()</signal>
   <receiver>LocationWidget</receiver>
   <slot>moveBottom()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>502</x>
     <y>912</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>497</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_btn_move_left</sender>
   <signal>clicked()</signal>
   <receiver>LocationWidget</receiver>
   <slot>moveLeft()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>385</x>
     <y>770</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>497</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_btn_move_right</sender>
   <signal>clicked()</signal>
   <receiver>LocationWidget</receiver>
   <slot>moveRight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>619</x>
     <y>770</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>497</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_btn_move_top</sender>
   <signal>clicked()</signal>
   <receiver>LocationWidget</receiver>
   <slot>moveTop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>502</x>
     <y>629</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>497</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_btn_rot_left</sender>
   <signal>clicked()</signal>
   <receiver>LocationWidget</receiver>
   <slot>rotateLeft()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>385</x>
     <y>629</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>497</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_btn_rot_right</sender>
   <signal>clicked()</signal>
   <receiver>LocationWidget</receiver>
   <slot>rotateRight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>619</x>
     <y>629</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>497</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setLocation(Punk::Engine::ILocation*)</slot>
  <slot>moveLeft()</slot>
  <slot>moveRight()</slot>
  <slot>moveTop()</slot>
  <slot>moveBottom()</slot>
  <slot>rotateRight()</slot>
  <slot>rotateLeft()</slot>
 </slots>
</ui>
